### PR TITLE
Replaced try/catch blocks with JUnit4's ExpectedException

### PR DESCRIPTION
**Pull request by a11n**
https://github.com/junit-team/junit/wiki/Rules#expectedexception-rules

### CODE REVIEW

File: src/test/java/com/jakewharton/fliptables/FlipTableTest.java [hunk 1]
```diff
@@ -146,41 +151,38 @@
...
-      FlipTable.of(headers, more);
-    } catch (IllegalArgumentException e) {
-      assertThat(e).hasMessage("Row 1's 4 columns != 2 columns");
-    }
+
+    exception.expect(IllegalArgumentException.class);
+    exception.expectMessage("Row 1's 1 columns != 2 columns");
+
+    String[][] less = { { "Less" } };
+    FlipTable.of(headers, less);
```
Comment (JakeWharton) [inline src/test/java/com/jakewharton/fliptables/FlipTableTest.java]: If this throws the test execution will not continue.

Comment (a11n) [inline src/test/java/com/jakewharton/fliptables/FlipTableTest.java]: Sorry, you are totally right. Is it okay for you to check both exceptions in separated methods?

Comment (JakeWharton) [inline src/test/java/com/jakewharton/fliptables/FlipTableTest.java]: Yep

Comment (a11n) [inline src/test/java/com/jakewharton/fliptables/FlipTableTest.java]: Cool! It's implemented in https://github.com/a11n/flip-tables/commit/abef402eba4da89131c517e88b50a4554234d97e. https://github.com/a11n/flip-tables/commit/aacecee7117298fe6fe9447a73ce4204001ad48a contains a slight adaption to meet current codestyle.

**Comment by JakeWharton** [comment]
Squashed and merged.

**Pull request closed without merging**